backend/danswer/search/pipeline.py

```diff
@@ -74,9 +86,73 @@ def __init__(
         self._relevant_section_indices: list[int] | None = None
 
         # Generates reranked chunks and LLM selections
-        self._postprocessing_generator: Iterator[
-            list[InferenceSection] | list[int]
-        ] | None = None
+        self._postprocessing_generator: (
+            Iterator[list[InferenceSection] | list[int]] | None
+        ) = None
+
+    def evaluate(
+        self, document: InferenceSection, query: str
+    ) -> dict[str, RelevanceChunk]:
+        relevance: RelevanceChunk = RelevanceChunk()
+        results = {}
+
+        # At least for now, is the same doucment ID across chunks
+        document_id = document.center_chunk.document_id
+        chunk_id = document.center_chunk.chunk_id
+
+        prompt = f"""
+        Analyze the relevance of this document to the search query:
+        Title: {document_id.split("/")[-1]}
+        Blurb: {document.combined_content}
+        Query: {query}
+
+        {AGENTIC_SEARCH_EVALUATION_PROMPT}
+        """
+
+        content = "".join(
```

Comment:
there's an invoke option which is better for when streaming is not needed

Reply:
Noted for future reference as well!